camelot/admin/action/list_action.py

```diff
@@ -399,17 +410,23 @@ class DeleteSelection( EditAction ):
     verbose_name = _('Delete')
 
     def gui_run( self, gui_context ):
-        #
-        # if there is an open editor on a row that will be deleted, there
-        # might be an assertion failure in QT, or the data of the editor 
-        # might be pushed to the row that replaces the deleted one
-        #
-        gui_context.item_view.close_editor()
-        super( DeleteSelection, self ).gui_run( gui_context )
-        # this refresh call could be avoided if the removal of an object
-        # in the collection through the DeleteObject action step handled this
-        gui_context.item_view.model().refresh()
-        gui_context.item_view.clearSelection()
+        if gui_context.item_view is not None:
+            #
+            # if there is an open editor on a row that will be deleted, there
+            # might be an assertion failure in QT, or the data of the editor
+            # might be pushed to the row that replaces the deleted one
+            #
+            gui_context.item_view.close_editor()
+            super( DeleteSelection, self ).gui_run( gui_context )
+            # this refresh call could be avoided if the removal of an object
+            # in the collection through the DeleteObject action step handled this
+            gui_context.item_view.model().refresh()
```

Comment:
deze refresh is mogelijk ook niet meer nodig, omdat de de DeleteObject action step dit nu denk ik wel afhandeld.

moest die wel nog nodig zijn, kunnen we die beter doen vdm een action step in de model_run, zodat er hier geen gui run meer nodig is

Reply:
done (was inderdaad niet meer nodig)